priam/src/main/java/com/netflix/priam/backup/BackupVerification.java

```diff
@@ -14,150 +14,60 @@
                         .collect(Collectors.toList());
-        logger.info("Snapshots found for {} : [{}]", result.selectedDate, backups);
-
-        // find the latest date (default) or verify if one provided
-        Date latestDate = null;
-        for (BackupMetadata backupMetadata : metadata) {
-            if (latestDate == null || latestDate.before(backupMetadata.getStart()))
-                latestDate = backupMetadata.getStart();
-
-            if (startTime != null
-                    && DateUtil.formatyyyyMMddHHmm(backupMetadata.getStart())
-                            .equals(DateUtil.formatyyyyMMddHHmm(startTime))) {
-                latestDate = startTime;
-                break;
-            }
-        }
-
-        result.snapshotTime = DateUtil.formatyyyyMMddHHmm(latestDate);
-        logger.info(
-                "Latest/Requested snapshot date found: {}, for selected/provided date: {}",
-                result.snapshotTime,
-                result.selectedDate);
-
-        // Get Backup File Iterator
-        String prefix = config.getBackupPrefix();
-        logger.info("Looking for meta file in the location:  {}", prefix);
-
-        Date strippedMsSnapshotTime = DateUtil.getDate(result.snapshotTime);
-        Iterator<AbstractBackupPath> backupfiles =
-                bkpStatusFs.list(prefix, strippedMsSnapshotTime, strippedMsSnapshotTime);
-        // Return validation fail if backup filesystem listing failed.
-        if (!backupfiles.hasNext()) {
-            logger.warn(
-                    "ERROR: No files available while doing backup filesystem listing. Declaring the verification failed.");
-            return result;
-        }
-
-        result.backupFileListAvail = true;
-
-        List<AbstractBackupPath> metas = new LinkedList<>();
-        List<String> s3Listing = new ArrayList<>();
-
-        while (backupfiles.hasNext()) {
-            AbstractBackupPath path = backupfiles.next();
-            if (path.getFileName().equalsIgnoreCase("meta.json")) metas.add(path);
-            else s3Listing.add(path.getRemotePath());
-        }
-
-        if (metas.size() == 0) {
-            logger.error(
-                    "No meta found for snapshotdate: {}", DateUtil.formatyyyyMMddHHmm(latestDate));
-            return result;
-        }
-
-        result.metaFileFound = true;
-        // Download meta.json from backup location and uncompress it.
-        List<String> metaFileList = new ArrayList<>();
-        try {
-            Path metaFileLocation =
-                    FileSystems.getDefault().getPath(config.getDataFileLocation(), "tmp_meta.json");
-            bkpStatusFs.downloadFile(Paths.get(metas.get(0).getRemotePath()), metaFileLocation, 5);
-            logger.info(
-                    "Meta file successfully downloaded to localhost: {}",
-                    metaFileLocation.toString());
+        metadata.sort((o1, o2) -> o2.getStart().compareTo(o1.getStart()));
```

Comment:
It was sorted reverse w.r.t start date (made the change to mention that explicitly). Since we have a lock on the snapshot job, one which starts last (and is successful) will be one which finishes last as well.